### PR TITLE
ut: fix HF_Echidna unit tests

### PR DESCRIPTION
**Pull request by shargon**
# Description

My bad, the previous code also checks in the executable folder.

## Type of change

- [ ] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Not changed

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Jim8y — APPROVED** [review]

**Comment by Jim8y** [comment]
Nice work @shargon

File: src/Neo/ProtocolSettings.cs [hunk 1]
```diff
@@ -124,6 +124,42 @@ public record ProtocolSettings
...
+
+            // Check if file exists in the specified (resolved) path
+            var fullPath = Path.Combine(path, fileName);
+            if (File.Exists(fullPath))
+            {
+                return fullPath;
+            }
+
+            // Check if file exists in the executable directory
+            var executableDir = AppDomain.CurrentDomain.BaseDirectory;
```
Comment (cschuchardt88) [inline src/Neo/ProtocolSettings.cs:152]: AppContent.BaseDirectory
```suggestion
            var executableDir = AppContext.BaseDirectory;
```

This works on all OSs and NetStandard

File: src/Neo/ProtocolSettings.cs [hunk 2]
```diff
@@ -143,7 +179,9 @@ public static ProtocolSettings Load(Stream stream)
         /// <returns>The loaded <see cref="ProtocolSettings"/>.</returns>
         public static ProtocolSettings Load(string path)
         {
-            if (!File.Exists(path))
+            path = FindFile(path, Environment.CurrentDirectory);
```
Comment (AnnaShaleva) [inline src/Neo/ProtocolSettings.cs:182]: My primary concern here is that this code should be compatible with the old behaviour of `neo-cli` binary for production usage (default `config.json` file from the `neo-cli` binary dir will be used). From what I see, it's OK, but we'd better recheck.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

**Pull request merged**